File: python/028. Implement strStr().py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        m = len(haystack)
        n = len(needle)
        if n > m: return -1
        if n == 0: return 0
        for i in range(m-n+1):
            flag = True
            for j in range(n):
                if haystack[i+j] != needle[j]:
                    flag = False
                    break
            if flag: return i
        return -1

File: C++/028. Implement strStr().cpp
class Solution {
public:
    bool compare(string& haystack, string& needle, int index)
    {
        for (int i = 0; i < needle.length(); i++)
        {
            if (haystack[index+i] != needle[i])
            {
                return false;
            }
        }
        return true;
    }
    int strStr(string haystack, string needle) 
    {
        int n = haystack.length();
        int m = needle.length();
        if(m > n) return -1;
        if (m == 0 && n == 0) return 0;
        for (int i = 0; i < n - m + 1; i++)
        {
            if (compare(haystack, needle, i))
            {
                return i;
            }
        }
        return -1;
    }
};
